l10n_it_pec_messages/model/mail_thread.py

```diff
@@ -100,30 +97,26 @@ def get_pec_attachments(self, cr, uid, message, context=None):
                         filename).strip()
                 else:
                     filename = decode(filename)
-            if (
-                filename == 'postacert.eml' or
-                filename == 'daticert.xml' or
-                filename == 'smime.p7s'
-            ):
-                if part.is_multipart() and len(part.get_payload()) > 1:
-                    raise orm.except_orm(
-                        _('Error'),
-                        _("Too many payloads for 'postacert.eml' or "
-                          "'daticert.xml'. Not handled"))
-                # http://goo.gl/zPRvxF
-                if part.is_multipart():
-                    # email.message.Message
-                    attachment = part.get_payload()[0]
-                else:
-                    # string
-                    attachment = part.get_payload(decode=True)
-                if filename == 'postacert.eml':
-                    postacert = attachment
-                if filename == 'daticert.xml':
-                    daticert = attachment
-                if filename == 'smime.p7s':
-                    smime = attachment
-        return (postacert, daticert, smime)
+            if num == 0 and n == 1 and part.get_content_type() == \
+                    'application/x-pkcs7-signature' and \
+                    filename == 'smime.p7s':
+                parts['smime.p7s'] = part.get_payload(decode=True)
+            elif num == 1 and n == 1 and part.get_content_type() == \
```

Comment:
Questo è vero, è molto restrittivo, me ne rendo conto. In generale credo che la modalità di composizione del messaggio sia sempre questa, però nel nostro caso il controllo su n potrebbe essere eliminato, visto i numerosi controlli che vengono già fatti sulla "profondità" del payload con num e sul suo content_type